packages/queries/src/util/iterateRecursively.ts

```diff
@@ -0,0 +1,53 @@
+// TODO: Unite/move this type with the one of the same
+// name already being used in `backend` and `queries`
+// See:
+// packages/backend/src/execution/executors/PgExecutor/queryBuilder/exp.ts
+// packages/queries/src/expression/Expression.ts
+type Primitive =
+    | string
+    | number
+    | boolean
+    | null
+    | undefined
+    | symbol
+    | bigint
+    | Date;
+
+type Traversable =
+    | Primitive
+    | { [key: string | number]: Traversable }
+    | Array<Traversable>;
+
+export function iterateRecursively<T extends Traversable>(
+    traversable: T,
+    visitor: (traversable: Traversable, path: string[]) => void,
+    path: string[] = [],
+): void {
+    // Apply the visitor to the current traversable
+    visitor(traversable, path);
+
+    if (
```

Comment:
if not a traversable thing, return early.

if it is a traversable thing, this can be simplified as follows:

```
Object.values(traversable).map(([key,item]) => iterateRecursively(...))
```